external/kafka/src/main/scala/org/apache/spark/streaming/kafka/DirectKafkaInputDStream.scala

```diff
@@ -84,6 +83,11 @@ class DirectKafkaInputDStream[
 
   protected var currentOffsets = fromOffsets
 
+  // Map to manage the time -> topic/partition+offset
+  private val offsetMap = new mutable.HashMap[Time, Map[TopicAndPartition, Long]]()
+  // Add to the listener bus for job completion hook
+  context.addStreamingListener(new DirectKafkaStreamingListener)
+
   @tailrec
   protected final def latestLeaderOffsets(retries: Int): Map[TopicAndPartition, LeaderOffset] = {
```

Comment:
Is there a reason to even add the streaming listener if the configuration option isn't turned on?  If the config option isn't on, couldn't you skip the listener and skip adding / removing items to the offset map altogether?